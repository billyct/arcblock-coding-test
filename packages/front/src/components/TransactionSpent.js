import {useState} from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

import Tooltip from './Tooltip'

const TransactionSpent = ({spent}) => {
  const [show, setShow] = useState(false)

  return (
    <Tooltip
      data-testid='TransactionSpent__tooltip'
      show={show}
      content={spent? 'Spent': 'Unspent'}
    >
      <div
        data-testid='TransactionSpent__div'
        onMouseEnter={() => setShow(true)}
        onMouseLeave={() => setShow(false)}
        className={cn('w-2 h-2 rounded-full', {
          'bg-green-500': spent,
          'bg-red-500': !spent,
        })}
      />
    </Tooltip>
  )
}

TransactionSpent.propTypes = {
  spent: PropTypes.bool,
}

TransactionSpent.defaultProps = {
  spent: false,
}

export default TransactionSpent
