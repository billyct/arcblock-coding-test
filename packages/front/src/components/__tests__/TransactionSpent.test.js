import {fireEvent, render} from '@testing-library/react'

import TransactionSpent from '../TransactionSpent'

jest.mock('../Tooltip', () => 'tooltip-component')

afterAll(() => {
  jest.resetModules()
})

describe('test TransactionSpent Component', () => {
  test('snapshot should correct', () => {
    const {container} = render(<TransactionSpent/>)
    expect(container).toMatchSnapshot()
  })

  test(`tooltip's show attribute should correct`, () => {
    const {getByTestId} = render(<TransactionSpent/>)

    const div = getByTestId('TransactionSpent__div')
    const tooltip = getByTestId('TransactionSpent__tooltip')

    fireEvent.mouseEnter(div)
    expect(tooltip).toHaveAttribute('show', 'true')

    fireEvent.mouseLeave(div)
    expect(tooltip).toHaveAttribute('show', 'false')
  })

  describe.each([
    [true, 'Spent', 'bg-green-500'],
    [false, 'Unspent', 'bg-red-500'],
  ])(`test attribute spent === %b`, (spent, expectedContent, expectedClassname) => {

    it(`should return ${expectedContent}`, function () {
      const {getByTestId} = render(<TransactionSpent spent={spent}/>)
      expect(getByTestId('TransactionSpent__tooltip')).toHaveAttribute('content', expectedContent)
    })

    it(`should return ${expectedClassname}`, function () {
      const {getByTestId} = render(<TransactionSpent spent={spent}/>)
      expect(getByTestId('TransactionSpent__div')).toHaveClass(expectedClassname)
    })
  })

})
