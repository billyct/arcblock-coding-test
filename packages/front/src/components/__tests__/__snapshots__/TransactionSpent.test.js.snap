// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TransactionSpent Component snapshot should correct 1`] = `
<div>
  <tooltip-component
    content="Unspent"
    data-testid="TransactionSpent__tooltip"
    show="false"
  >
    <div
      class="w-2 h-2 rounded-full bg-red-500"
      data-testid="TransactionSpent__div"
    />
  </tooltip-component>
</div>
`;
